src/keyring.ts

```diff
@@ -44,8 +44,9 @@ interface FullKeySet {
 }
 
 function deriveKeySet(seed: Uint8Array, v03ID?: string): FullKeySet {
-  let hdNode = HDNode.fromSeed(seed).derivePath(BASE_PATH)
-  if (v03ID) hdNode = hdNode.derivePath(ROOT_STORE_PATH)
+  const seedNode = HDNode.fromSeed(seed)
+  let hdNode = seedNode.derivePath(BASE_PATH)
+  if (v03ID) hdNode = seedNode.derivePath(LEGACY_BASE_PATH)
```

Comment:
`derivePath` is a very expensive operation! Please make it happen only once if it's a `v03ID`.